### DISCUSSION TITLE
Fundamentos do Entity Framework > Introdução > Create  |  Sobre Data Annotations

### DISCUSSION THREAD

**Discussion by DrenokBR** [balta-io/forum#426] [Tira Dúvidas] [answered]
Entre 2 min 0 seg e 2m 4s da aula por que o "Data Annotation" que você usou na Classe Tag é [Table("Tag")]
e não [Table("Tags")] com a palavra "Tags" no plural ?!?! O problema da implementação para adicionar um registro na tabela "Tag" do DB não se dá pelo fato de estar usando a palavra no plural (context.Tags.Add(Tag)) ???

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @DrenokBR, como vai?
É uma questão de gosto... eu prefiro no singular, mas pode ser no plural sim.

A ideia do exemplo, é mostrar como você pode nomear uma tabela, e isto é muito importante.
Quando deixamos o EF nomear, ele vai colocar no plural, mas do inglês... Então, "produto" ficaria "produtoes" já que no inglês palavras no singular que terminam com "o" recebem "es" como sufixo.

**Reply by DrenokBR** [reply]
Bom Dia !
Mas o que eu não entendi foi o fato da causa do problema ter sido o código " (context.*Tags*.Add(Tag)) " estar com a palavra " *Tags* " no plural e a solução com "Data Annotation" estar com a palavra no singular. Eu venho estudando C# e .NET por outros cursos, mas não cheguei a ver nenhuma aula sobre "Data Annotations". Talvez seja esse o motivo da minha dificuldade.

**Reply by andrebaltieri** [reply]
@DrenokBR são coisas distintas... o `context.Tags` se refere ao DbSet, a propriedade que você criou no DbContext (Representação do banco de dados na memória) e que representa uma lista de tags.

Desta forma, faz total sentido o nome da propriedade chamar tags:
```csharp
public class AppDbContext : DbContext
{
    public DbSet<Tag> Tags { get; set; }
}

...

context.Tags.Add(...);
```

Os `DataAnnotations` são uma forma de decorar as classes e propriedades para que não haja necessidade de usar os mapeamentos (Mapping) do EF.

Ficou mais claro? Qualquer dúvida posta aí

**Reply by DrenokBR** [reply]
Aos 1m 36 s da aula *Create* > *Introdução* > *Fundamentos do Entity Framework* você diz que o código " context.*Tags*.Add(Tag) " dentro da classe Program não vai funcionar porque o EF vai procurar uma tabela no DB com o nome *Tags* e não existe uma tabela com esse nome. Aí você vai lá e coloca o Data Annotation [Table("*Tag*")] na classe *Tag*. Por que isso resolve o problema ?! Como isso funciona ?! Se o EF vai procurar uma tabala *Tags* o Data Annotation não deveria ser [Table("*Tags*")]  ?! Isso que não entendi.

**Reply by DrenokBR** [reply]
Dei uma pesquisada por aqui e agora acho que entendi. Veja se é isso : O Data Annotation "Table" vai "informar" ao EF que todo objeto criado da classe *Tag* estará associada a Tabela "Tag" do DB, ou seja, o Data Annotation Table vai "forçar" esta associação para o EF.

É isso, né ?

**Reply by andrebaltieri** [reply]
Isso ai!!! 🚀
